backend/app/rest/residents_routes.py

```diff
@@ -21,6 +21,16 @@ def add_resident():
             400,
         )
 
+    # Check for the existence of a resident prior to adding them
+    fmt_resident_id = resident.get("initial") + str(resident.get("room_num"))
+    try:
+        res = residents_service.get_residents(False, 1, 10, fmt_resident_id)
+        if len(res["residents"]) > 0:
```

Comment:
When adding a new resident, we simply want to check for a result length greater than 0 (if there's even one other resident that has the same ID, then this is a duplicate resident => reject)